Posts/2021/04/NewTemporaryFoldersPowerShell

```diff
@@ -0,0 +1,89 @@
+---
+post_title: Borrowing a Built in PowerShell to create a Temporary Folder
+username: sean.kearney@microsoft.com
+Catagories: PowerShell, Function
+tags: Function,Fun trick,Existing Cmdlet,New Purpose
+Summary: Leveraging Built in Cmdlets in a New and interesting ways
+---
+
+**Q:** Hey question for you.  It seems silly and I know I could probably put something together with Get-Random.  But can you think of another way to create a temporary folder with a random in PowerShell? Ideally I'd like it to be in a user's own "Temporary Folder" is possible.
+
+**A:**  We sure can!  If Doctor Scripto was sitting here right now I'd see that little green haired shout out "Never fear, Scripto is here!"
+
+
+## New-TemporaryFile Cmdlet
+
+Within PowerShell there is a built in Cmdlet called “New-TemporaryFile”.  Running this Cmdlet simply creates a random 0 byte file in the $ENV:Temp folder in whichever platform you are working in.
+
+However, we can “Borrow” the filename created and use it to create a folder instead.   It’s not really difficult, but maybe just not thought of very often.
+
+`New-TemporaryFile`
+
+It will create a file like below.
+
+Mode                 LastWriteTime         Length Name
+----                 -------------         ------ ----
+-a----         3/31/2021   9:25 PM              0 tmpA927.tmp
+
+ 
+Ok, that really wasn’t that impressive but what if we were to do this instead?
+
+`$File=New-TemporaryFile`
+
+Now we’ve created the file and stored it away in the $File object.   With this we can remove the File of course using the Remove-Item Cmdlet
+
+`Remove-Item -path $File -force`
+
+AHA!  I’ve all of that hard work.  But the $File object is still there to use.  So I could access the name and use it to create a directory instead in the following manner
```

Comment:
HA!  I’ve done all of that hard work, but the `$File` object is still there to use.  So I could access the name and use it to create a directory instead in the following manner:

Reply:
Use single spaces between sentences.

Again, put the code in a code block. Don't use inline code markup unless it is part of a paragraph. Code alone like this should always be in a fenced code block.